src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,150 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private T value;
+    private int index;
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value);
+        if (head == null) {
+            head = newNode;
+            newNode.prev = tail;
+        } else {
+            newNode.prev = tail;
+            tail.next = newNode;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> newNode = new Node<>(value);
+        Objects.checkIndex(index, size + 1);
```

Comment:
Please, first check common mistakes in this task and fix them, after that, mentor will review it again. Thanks!

Reply:
Don’t use class Objects.